examples/distance.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2018 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * Example program that calculates the distance in kilometers between two
+ * hexagon indices.
+ */
+
+// mean Earth radius
+#define R 6371.0088
+
+#include <h3/h3api.h>
+#include <math.h>
+#include <stdio.h>
+
+/**
+ * @brief haversineDistance finds the
+ * [great-circle distance](https://en.wikipedia.org/wiki/Great-circle_distance)
+ * between two points on a sphere.
+ * @see https://en.wikipedia.org/wiki/Haversine_formula.
+ *
+ * Parameters are the latitude and longitude of the first and second point in
+ * radians, respectively.
+ *
+ * @return the great-circle distance in kilometers.
+ */
+double haversineDistance(double th1, double ph1, double th2, double ph2) {
+    double dx, dy, dz;
+    ph1 -= ph2;
+
+    dz = sin(th1) - sin(th2);
+    dx = cos(ph1) * cos(th1) - cos(th2);
+    dy = sin(ph1) * cos(th1);
+    return asin(sqrt(dx * dx + dy * dy + dz * dz) / 2) * 2 * R;
+}
+
+int main(int argc, char *argv[]) {
+    // 1455 Market St @ resolution 15
+    H3Index h3HQ1 = stringToH3("8f2830828052d25");
+    // 555 Market St @ resolution 15
+    H3Index h3HQ2 = stringToH3("8f283082a30e623");
+
+    double distance = 0;
```

Comment:
Looks like `double distance` is not used.

Reply:
uh oh, possible to configure linter to catch?  compiling with `-Wall` catches this...

Reply:
> publicizing geoDistRads/geoDistKm

I suggest we put out a minor version feature bump with these two and `h3DistRads/h3DistKm`; doing that is out of scope for this addition. like I mentioned to @nrabinowitz , I expect the h3 distance functions will be more useful and appropriate than yet another set of geo distance functions.